lib/web/command_utils.go

```diff
@@ -130,3 +137,63 @@ func (s *syncRWWSConn) ReadMessage() (messageType int, p []byte, err error) {
 	defer s.rmtx.Unlock()
 	return s.WSConn.ReadMessage()
 }
+
+func newBufferedPayloadWriter(pw *payloadWriter, buffer *summaryBuffer) *bufferedPayloadWriter {
+	return &bufferedPayloadWriter{
+		payloadWriter: pw,
+		buffer:        buffer,
+	}
+}
+
+type bufferedPayloadWriter struct {
+	*payloadWriter
+	buffer *summaryBuffer
+}
+
+func (bp *bufferedPayloadWriter) Write(data []byte) (int, error) {
+	bp.buffer.Write(bp.nodeID, data)
+	return bp.payloadWriter.Write(data)
+}
+
+func newSummaryBuffer(capacity int) *summaryBuffer {
+	return &summaryBuffer{
+		buffer:            make(map[string][]byte),
+		remainingCapacity: capacity,
+		invalid:           false,
+		mutex:             sync.Mutex{},
+	}
+}
+
+type summaryBuffer struct {
+	buffer            map[string][]byte
+	remainingCapacity int
+	invalid           bool
+	mutex             sync.Mutex
```

Comment:
nit: Can you add a comment and mention that this mutex is protecting all members of this struct?